changelog/unreleased/features/2268--compress-table-slice.md

```diff
@@ -0,0 +1,5 @@
+The `segment-store` store backend now compresses data before persisting it,
+resulting in 5x space savings for newly archived data, and an up to X% memory
+usage reduction. Depending on the speed of the disk this may improve the overall
+speed of VAST as well; we measured a negligible less than 1% performance penalty
+using the fastest disks we have (5GB/s).
```

Comment:
Note that X = 10% on my local M1 Max MBP 16' for a 100 GB database; I left this running earlier today.

We should roll this PR out on the testbed to get some numbers such that we can put them in here. I also measured with `vast.import.batch-size: 65536` rather than the default `1024` because I forgot to revert to the defaults, but quick local tests show that the numbers are almost as good with the measly default batch size.

Reply:
> We should note that compression trades random-access performance.

That's not necessarily true. The compression is per column, and once you access it it actually stays decompressed in-memory in the record batch until the ref count drops to zero.

> In fact, if the archive is now smaller than the index, we should make dense indexes optional as well in the same release.

I disagree—that would be follow-up work. There's no reason to delay an improvement just because there's room for further improvement because of it. By that logic we would never be able to cut a release at all.

I also doubt that we can deduce from the relative size of store and dense indexes alone that the dense indexes are now irrelevant. While I agree that it is likely, we would need benchmarks confirming this before making that change. Please write a follow-up story if you want to investigate this further.

Reply:
Note that I've rewritten the changelog entry since we started this discussion to also show the overall reduction in space and to also mention the tuned defaults.